kv/range_cache.go

```diff
@@ -195,18 +203,21 @@ func (rmc *rangeDescriptorCache) getCachedRangeDescriptor(key proto.Key) (
 	rangeCacheKey, *proto.RangeDescriptor) {
 	rmc.rangeCacheMu.RLock()
 	defer rmc.rangeCacheMu.RUnlock()
-	return rmc.getCachedRangeDescriptorLocked(key)
+	return rmc.getCachedRangeDescriptorLocked(key, false)
 }
 
 // getCachedRangeDescriptorLocked is a helper function to retrieve the
 // descriptor of the range which contains the given key, if present in the
 // cache. It is assumed that the caller holds a read lock on rmc.rangeCacheMu.
-func (rmc *rangeDescriptorCache) getCachedRangeDescriptorLocked(key proto.Key) (
+func (rmc *rangeDescriptorCache) getCachedRangeDescriptorLocked(key proto.Key, inclusive bool) (
 	rangeCacheKey, *proto.RangeDescriptor) {
-	// We want to look up the range descriptor for key. The cache is
-	// indexed using the end-key of the range, but the end-key is
-	// non-inclusive. So we access the cache using key.Next().
-	metaKey := keys.RangeMetaKey(key.Next())
+	// The cache is indexed using the end-key of the range, but the
+	// end-key is non-inclusive. If inclusive is false, we access the
```

Comment:
This comment is implementation-focused; could you add a comment outside the function explaining this from the caller's perspective (i.e. what it means to pass inclusive or non-inclusive, because it's hard to tell from the outside why one caller passes true and the others don't)